tutor/utils.py

```diff
@@ -163,11 +163,19 @@ def docker(*command: str) -> int:
 
 
 def docker_compose(*command: str) -> int:
-    if shutil.which("docker-compose") is None:
+    """
+    Fallback to docker compose when docker-compose is not available
+    """
+    if shutil.which("docker-compose") is not None:
+        return execute("docker-compose", *command)
+    elif "is not a docker command" not in str(
```

Comment:
I understand that you are doing this to make sure that docker-compose v2 is installed, but we should not run this command for every call to `docker-compose`. 

Also, I'd like to be able to force the use of docker-compose v2 by defining an environment variable; this would be useful for me to troubleshoot v2 issues.

So, here's what I propose:

- If `TUTOR_USE_DOCKER_COMPOSE_V2` is in `os.environ`, use `docker compose`.
- Else, if "docker-compose" is present, use it.
- Else, check for `docker compose version` and cache the result with [lru_cache](https://docs.python.org/3.8/library/functools.html#functools.lru_cache). If present, use `docker compose`. (don't check for the presence of "is not a docker command", but check the result `returncode` instead)
- Else, raise TutorError.

Also, please add an entry to the changelog, following: https://docs.tutor.overhang.io/tutor.html#contributing

Reply:
> Why is an env variable preferred over a variable in config.yml

config.yml files are meant to be portable from one server to the next, despite the fact that they may not all run the same docker-compose version.